py/dml/codegen.py

```diff
@@ -1207,6 +1207,13 @@ def expr_variable(tree, location, scope):
         if in_dev_tree:
             e = in_dev_tree
     if e is None:
+        # TODO/HACK: The discard ref is exposed like this to allow it to be as
+        # keyword-like as possible while still allowing it to be shadowed.
+        # Once we remove support for discard_ref_shadowing the discard ref
+        # should become a proper keyword and its codegen be done via dedicated
+        # dispatch
+        if name == '_' and tree.site.dml_version() != (1, 2):
```

Comment:
This may not suffice as a reminder, since the comment is not adjacent to the code that will be touched when the compat is removed. One may argue that you mention discard_ref_shadowing in the comment so grep will find it, but you may consider a `# when removing, remember to also fix ..` comment near the enabled_compat-checking code.

Reply:
I was thinking of the code in dmlparse that checks enabled_compat, but I suppose compat.py makes even more sense.